frontend/src/components/Map/Map.js

```diff
@@ -0,0 +1,397 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import ReactDOM from 'react-dom'
+import mapboxgl from 'mapbox-gl'
+import { Box, createStyles, GlobalStyles, Tooltip } from '@mui/material'
+import { InfoOutlined } from '@mui/icons-material'
+import SimpleRange from '../SimpleRange'
+import Select from '../Select'
+import MapPopupContent from '../MapPopupContent'
+import { buildColorExpression, createMapConfig } from '../../utils/helpers'
+import { MAPBOX_API_KEY } from '../../utils/constants'
+
+mapboxgl.accessToken = MAPBOX_API_KEY
+
+const popupClassName = 'floating-popup'
+const popupContainerStyles = createStyles({
+  '.mapboxgl-popup-tip': {
+    display: 'none',
+  },
+
+  '.mapboxgl-popup-content': {
+    padding: '0',
+    background: 'transparent',
+    boxShadow: 'none',
+  },
+
+  [`.${popupClassName}.immobile .mapboxgl-popup-content`]: {
+    background: 'rgba(255, 255, 255, 1.0)',
+  },
+})
+
+function Map({ config }) {
+  const mapContainer = useRef(null)
+  const map = useRef(null)
+  const clearMapFunctionsList = useRef([])
+  const [selectedMetric, setSelectedMetric] = useState('')
+  const [data, setData] = useState([])
+  const [mapLoaded, setMapLoaded] = useState(false)
+
+  const metricData = useMemo(
+    () => (config?.metrics || [])?.find(item => item?.id === selectedMetric),
+    [config?.metrics, selectedMetric],
+  )
+
+  const colorScheme = useMemo(
+    () => metricData?.layers?.[0]?.paint?.default?.fill?.colorScheme || [],
+    [metricData?.layers],
+  )
+  const colorSchemeReversed = useMemo(() => [...colorScheme].reverse(), [colorScheme])
+
+  // methods
+  const getData = useCallback(async () => {
+    setData([])
+    const url = metricData?.data?.url
+    if (!url) return
+
+    // TODO: use client instance instead
+    try {
+      const res = await fetch(metricData?.data?.url)
+      const json = await res.json()
+      setData(json)
+    } catch (e) {
+      console.error(e)
+    }
+  }, [metricData?.data?.url])
```

Comment:
**getData** should be using useCallback as it's a dependency for that effect, and causes infinite loop when using arrow function as it's being recreated on every render.
**Memoizing** of vars is done for the same reason, just to prevent updating of effects and callbacks